app/models/supporter.rb

```diff
@@ -143,22 +202,35 @@ def filled_address_fields?
   end
 
   def update_primary_address
-    if self.changes.slice(*ADDRESS_FIELDS).any? #changed an address field
-      if filled_address_fields?
-        if primary_address.nil?
-          self.addresses.build(address_field_attributes)
-        else
-          primary_address.update(address_field_attributes)
-        end
-      elsif primary_address.present?
-        prim_addr = primary_address
-        self.update(primary_address: nil)
-        self.addresses.delete(prim_addr)
-        prim_addr.destroy
+    if filled_address_fields?
+      if primary_address.nil?
+        self.addresses.build(address_field_attributes)
+      else
+        primary_address.update(address_field_attributes)
       end
+    elsif primary_address.present? && empty_new_address_fields?
+      delete_primary_address
     end
   end
 
+  def empty_new_address_fields?
+    return false if address.present?
+    return false if city.present?
+    return false if state_code.present?
+    return false if country.present?
+    return false if zip_code.present?
+    true
+  end
```

Comment:
This checks if the address is being updated to `nil` fields. Do you have any suggestions on how to do this in a better way?

Reply:
Would just returning `address.blank? && city.blank? && state_code.blank? && country.blank? && zip_code.blank?` work?

Reply:
Yes :laughing: